### ISSUE TITLE
Multi-machine Vagrantfile has odd vm.provider behavior

### ISSUE THREAD

**Issue by sparrc** [mitchellh/vagrant#7893]
### Vagrant version

1.8.6

### Host operating system

Mac Sierra

### Guest operating system

Ubuntu & Windows

### Vagrantfile
```ruby
VAGRANTFILE_API_VERSION = "2"

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
  config.vm.define "ubuntu" do |ubuntu|
    ubuntu.vm.box = "ubuntu/trusty64"
    ubuntu.vm.provider "virtualbox"
  end

  config.vm.define "windows" do |boo|
    boo.vm.box = "opentable/win-2012r2-standard-amd64-nocm"
    boo.vm.provider "vmware_fusion"
  end
end
```

### Expected behavior

In the above config, I expect that the "ubuntu" box would use the virtualbox provider, and the "windows" box would use the vmware_fusion provider.

### Actual behavior

they both use vmware_fusion.

Strangely, if I add another vmware_fusion provider config line to the "ubuntu" box after the virtualbox provider, then it works as intended:

```ruby
VAGRANTFILE_API_VERSION = "2"

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
  config.vm.define "ubuntu" do |ubuntu|
    ubuntu.vm.box = "ubuntu/trusty64"
    ubuntu.vm.provider "virtualbox"
    ubuntu.vm.provider "vmware_fusion"
  end

  config.vm.define "windows" do |boo|
    boo.vm.box = "opentable/win-2012r2-standard-amd64-nocm"
    boo.vm.provider "vmware_fusion"
  end
end
```

**Comment by sparrc** [comment]
Note that on this system, my default vm provider is vmware_fusion. It appears that maybe the issue is that the vm.provider config does not override the default provider unless the default provider is also specified after the desired provider.

To me the expected behavior should be that vm.provider overrides the default provider.

**Comment by chrisroberts** [comment]
Thanks for reporting this issue. This was fixed via #7756 and should work as expected in the next release. Cheers!
